pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/ConcurrentMapPartitionUpsertMetadataManager.java

```diff
@@ -181,6 +210,29 @@ protected void removeSegment(IndexSegment segment, MutableRoaringBitmap validDoc
     }
   }
 
+  /**
+   * When TTL is enabled for upsert, this function is used to remove expired keys from the primary key indexes.
+   * This function will be called before new consuming segment start to consume.
+   *
+   * @param watermark The watermark is the time used to clean up the metadata in the previous round
```

Comment:
Can you define what is a watermark? Is it about time? what is the unit?

Reply:
The watermark is the time used to clean up the metadata in the previous round. 
it's millisecond unixtimestamp in long value